docs/03-code-internals/19-service-objects.md

```diff
@@ -0,0 +1,601 @@
+---
+title: Using service objects at Discourse
+short_title: Service objects
+id: service-objects
+---
+# Overview
+
+A service is a small object that encompasses business logic for a given action.
+From outside, it should be seen as a sort of black box. You provide it with parameters, it runs, and then it returns a result object describing the outcome of the action.
```

Comment:
Is it worth calling out that a service object can have side effects here?

Reply:
I changed the sentence to “[…] it runs (including all the side effects it can trigger) […]”.